addons/account/tests/test_invoice_taxes.py

```diff
@@ -261,6 +261,76 @@ def test_tax_repartition(self):
         self.assertAlmostEqual(abs(ref_tax_lines.filtered(lambda x: x.account_id == account_1).balance), 37.8, 2, "Refund tax line on account 1 should amount to 37.8 (90% of 42)")
         self.assertEqual(ref_tax_lines.mapped('tag_ids'), ref_tax_tag, "Refund tax lines should have the right tag")
 
+    def test_division_tax(self):
+        '''
+        Test that when using division tax, with percentage amount
+        100% any change on price unit is correctly reflected on
+        the whole move.
+
+        Complete scenario:
+            - Create a division tax, 100% amount, included in price.
+            - Create an invoice, with only the mentioned tax
+            - Change price_unit of the aml
+            - Total price of the move should change as well
+        '''
+
+        sale_tax = self.env['account.tax'].create({
+            'name': 'tax',
+            'type_tax_use': 'sale',
+            'amount_type': 'division',
+            'amount': 100,
+            'price_include': True,
+            'include_base_amount': True,
+            'invoice_repartition_line_ids': [
+                (0, 0, {
+                    'repartition_type': 'base',
+                    'factor_percent': 100.0,
+                    'tag_ids': [(6, 0, self.base_tag_pos.ids)],
+                }),
+                (0, 0, {
+                    'repartition_type': 'tax',
+                    'factor_percent': 100.0,
+                    'tag_ids': [(6, 0, self.tax_tag_pos.ids)],
+                }),
+            ],
+            'refund_repartition_line_ids': [
+                (0, 0, {
+                    'repartition_type': 'base',
+                    'factor_percent': 100.0,
+                    'tag_ids': [(6, 0, self.base_tag_neg.ids)],
+                }),
+                (0, 0, {
+                    'repartition_type': 'tax',
+                    'factor_percent': 100.0,
+                    'tag_ids': [(6, 0, self.tax_tag_neg.ids)],
+                }),
+            ],
```

Comment:
if you don't care about `invoice_repartition_line_ids` / `refund_repartition_line_ids`, you can pop them from the dictionary